docs/zh/maintain/diagnose.md

```diff
@@ -2,35 +2,119 @@
-    path: /work/openmldb
+### status 状态
+
+`status`用于查看OpenMLDB集群的状态，包括服务组件的地址，角色，连接时间，状态等，等价于`SHOW COMPONENTS`。如果发现集群表现不正常，请先查询各服务组件的实时状态。
 ```
+openmldb_tool status -h
+usage: openmldb_tool status [-h] [--helpfull] [--diff]
+
+optional arguments:
+  -h, --help  show this help message and exit
+  --helpfull  show full help message and exit
+  --diff      check if all endpoints in conf are in cluster. If set, need to set `--conf_file`
+```
+
+- 简单查询集群状态：
+  ```
+  openmldb_tool status [--cluster=...]
+  ```
+  输出类似下表：
+  ```
+  +-----------------+-------------+---------------+--------+---------+
+  |     Endpoint    |     Role    |  Connect_time | Status | Ns_role |
+  +-----------------+-------------+---------------+--------+---------+
+  | localhost:10921 |    tablet   | 1677398926974 | online |   NULL  |
+  | localhost:10922 |    tablet   | 1677398926978 | online |   NULL  |
+  |  localhost:7527 |  nameserver | 1677398927985 | online |  master |
+  |  localhost:9902 | taskmanager | 1677398934773 | online |   NULL  |
+  +-----------------+-------------+---------------+--------+---------+
+  ```
+
+#### 检查配置文件与集群状态是否一致
+
+如果指定`--diff`参数，会检查配置文件中的所有节点是否都在已经启动的集群中，如果有节点不在集群中，会输出异常信息。如果集群中有节点不在配置文件中，不会输出异常信息。需要配置`-f,--conf_file`，例如，你可以在镜像里这样检查：
+```bash
+openmldb_tool status --diff -f=/work/openmldb/conf/hosts
+```
+
+### inspect 检查
+
+`inspect`用于检查集群的在线和离线两个部分是否正常工作，可以选择单独检查`online`或`offline`，不指定则都检查。可以定期执行检查，以便及时发现异常。
+```
+openmldb_tool inspect -h
+usage: openmldb_tool inspect [-h] [--helpfull] {online,offline} ...
+
+positional arguments:
+  {online,offline}
+    online          only inspect online table
+    offline         only inspect offline jobs, check the job log
+```
+在线检查会检查集群中的表状态（包括系统表），并输出有异常的表，包括表的状态，分区信息，副本信息等，等价于`SHOW TABLE STATUS`并筛选出有异常的表。如果发现集群表现不正常，请先检查下是否有异常表。例如，`SHOW JOBS`无法正常输出历史任务时，可以`inspect online`检查一下是否是job系统表出现问题。
+
+离线检查会检查集群中的离线任务，并输出最终状态为失败的任务（不检查“运行中”的任务）及其日志，等价于`SHOW JOBS`并筛选出失败任务。
+
+### test 测试
+
+`test`将执行一些测试SQL，包括：创建库，创建表，在线插入数据，在线查询数据，删除表，删除库。如果TaskManager组件存在，还会执行离线查询任务，由于没有做离线导入数据，查询结果理应为空。
+
+在新建好集群时或发现集群表现不正常时，可以执行`test`测试集群是否能工作。
+
+### static-check 静态检查
 
-集群版yaml
+`static-check`静态检查，根据集群部署配置文件（通过参数`-f,--conf_file`指定），登录各个服务组件的部署地址，可以收集版本信息、配置文件、日志文件，检查版本是否一致，对收集到的配置文件和日志文件做分析。可以在集群未部署前进行检查，避免因程序版本或配置文件错误导致的集群部署失败。或在集群异常时，将分布式的日志文件收集在一起，方便调查问题。
+
+```bash
+openmldb_tool static-check -h
+usage: openmldb_tool static-check [-h] [--helpfull] [--version] [--conf] [--log]
+
+optional arguments:
+  -h, --help     show this help message and exit
+  --helpfull     show full help message and exit
+  --version, -V  check version
+  --conf, -C     check conf
+  --log, -L      check log
+```
+
+#### 部署配置文件
+
+`-f,--conf_file`部署配置文件格式可以是hosts风格或yaml风格，描述集群中有哪些组件，分布在哪个节点以及部署目录。
+- hosts风格（参考release包中的`conf/hosts`）
+```
+[tablet]
+localhost:10921 /tmp/openmldb/tablet-1
+localhost:10922 /tmp/openmldb/tablet-2
+
+[nameserver]
+localhost:7527
+
+[apiserver]
+localhost:9080
+
+[taskmanager]
+localhost:9902
```

Comment:
the default deploy directory is `OPENMLDB_HOME`, which is defaulted to the current openmldb release directory.

```
OPENMLDB_HOME="$(cd "$(dirname "$0")"/.. || exit 1; pwd)"
```

Alternatively, we can update the deploy scripts to generate a hosts_deployed with all the detailed informations.

Reply:
We can set the default deploy dir by `--default_dir` in diag tool, users can set it manually now, default value is `/work/openmldb` for demo docker. `hosts_deployed` is a great idea, we can improve in the future